### ISSUE TITLE
[feature request] [terraform-provider-aws] Allow specifying the credentials profile to use

### ISSUE THREAD

**Issue by mtougeron** [hashicorp/terraform#4670]
When using the `provider "aws" {}` it uses the `default` profile from `~/.aws/credentials` (or OS specific path). While you are able to set the environment variable `AWS_PROFILE` it would be much easier to allow setting the profile to use as part of the provider definition.

This would allow us to do something like this and is a lot less prone to error.

```
variable "aws_account_ids" {
  default = {
    "dev"   = "12345"
    "qa"    = "12345"
    "prod"  = "09876"
    "stage" = "09876"
  }
}

variable "aws_profile_map" {
  default = {
    "dev"   = "nonprod"
    "qa"    = "nonprod"
    "prod"  = "prod"
    "stage" = "prod"
  }
}

provider "aws" {
  profile = "${lookup(var.aws_profile_map, var.env)}"

  allowed_account_ids = ["${lookup(var.aws_account_ids, var.env)}"]
}
```

**Comment by jen20** [comment]
Hi @mtougeron. I actually implemented this feature the other day in #4632, but the test coverage needs some attention. This is also being tracked in #2235. Given that, I'm going to close this issue for now - if you subscribe to the pull request you'll get notified when the feature is implemented. Thanks for opening the issue!

**Comment by mtougeron** [comment]
@jen20 Awesome!
